### PR TITLE
Improve native auth input box

### PR DESCRIPTION
**Pull request by CiprianDraghici**

### CODE REVIEW

File: src/pages/Authentication/components/Input/components/Textarea/Textarea.tsx [hunk 1]
```diff
@@ -1,104 +1,39 @@
...
-import { TokenColorsEnum } from 'pages/Authentication/enum';
-import type { TextareaDivisionType } from './types';
-import styles from './styles.module.scss';
 import { useGetNativeAuthToken } from 'hooks/useGetNativeAuthToken';
 import { FormValuesType } from '../../types';
 import { useTokenActions } from '../../hooks/useTokenActions';
 import { CopyButton } from '@multiversx/sdk-dapp/UI/CopyButton';
-
-const DEFAULT_COLOR = '#000000';
+import CodeEditor from '@uiw/react-textarea-code-editor';
```
Comment (arhtudormorar) [inline src/pages/Authentication/components/Input/components/Textarea/Textarea.tsx:7]: eslint is not working properly. this should come before relative imports by default

File: src/pages/Authentication/components/Input/hooks/useTokenActions.ts [hunk 2]
```diff
@@ -7,7 +7,8 @@ import { useChain } from 'hooks/useChain';
 import { decodeToken } from '../components/Textarea/helpers/decodeToken';
 import { validateToken } from '../components/Textarea/helpers/validateToken';
 import { FormValuesType } from '../types';
-import { MetricType } from '../../../types';
+import { MetricType } from 'pages/Authentication/types';
+import debounce from 'lodash.debounce';
```
Comment (arhtudormorar) [inline src/pages/Authentication/components/Input/hooks/useTokenActions.ts:11]: same observation

File: src/pages/Authentication/components/Input/hooks/useTokenActions.ts [hunk 3]
```diff
@@ -92,32 +68,47 @@ export const useTokenActions = () => {
...
-      const allowedCombinations =
-        commonKey && allowedCombinationKeys.includes(e.key);
-      const allowed =
-        commonKey || arrowKeys.includes(e.key) || allowedCombinations;
+      setFieldValue('token', token, false);
+      setFieldTouched('token', true, false);
 
-      if (!allowed) {
-        handlePreventDefault(e);
+      if (!token || token.match(/[\r\n]/gm) || token.match(/\s+/g)) {
```
Comment (arhtudormorar) [inline src/pages/Authentication/components/Input/hooks/useTokenActions.ts:86]: I find it hard to read token.match(/[\r\n]/gm), maybe isolate it into a constant?

File: src/pages/Authentication/components/Input/components/Textarea/Textarea.tsx (hunk 1 above)
Comment (CiprianDraghici) [reply src/pages/Authentication/components/Input/components/Textarea/Textarea.tsx:7]: to fix the `eslint` rules is not in the scope of this PR, but certainly we have to fix it

File: src/config.tsx [hunk 4]
```diff
@@ -10,3 +10,5 @@ export const ApiBaseUrls = {
   [EnvironmentsEnum.devnet]: 'https://devnet-tools.multiversx.com',
   [EnvironmentsEnum.testnet]: 'https://testnet-tools.multiversx.com'
 };
+
+export const miscApi = 'https://misc-api.multiversx.com';
```
Comment (arhtudormorar) [inline src/config.tsx:14]: MISC_API ?

File: src/pages/Authentication/components/Input/components/Textarea/plugins/splitToken.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
+import { Element, Literal } from 'hast';
+
+export const splitToken = () => {
```
Comment (arhtudormorar) [inline src/pages/Authentication/components/Input/components/Textarea/plugins/splitToken.ts:3]: add basic tests

File: src/pages/Authentication/context/AuthenticationContext.ts [hunk 6]
```diff
@@ -98,5 +114,39 @@ export const useAuthenticationValue = () => {
     [metrics]
   );
 
-  return { metrics, setMetrics, metricItems };
+  const fetchInitialTokens = async () => {
+    try {
+      const { data } = await axios.get<InitialTokensType>(
+        `${miscApi}/utils-native-auth`
+      );
+
```
Comment (arhtudormorar) [inline src/pages/Authentication/context/AuthenticationContext.ts:122]: setInitialTokens(value);
        setFetchingInitialTokens(false);

directly

File: src/pages/Authentication/context/AuthenticationContext.ts [hunk 7]
```diff
@@ -98,5 +114,39 @@ export const useAuthenticationValue = () => {
...
+      return data;
+    } catch (err) {
+      console.error(err);
```
Comment (arhtudormorar) [inline src/pages/Authentication/context/AuthenticationContext.ts:125]: setFetchingInitialTokens(false);

File: src/config.tsx (hunk 4 above)
Comment (CiprianDraghici) [reply src/config.tsx:14]: Used the same naming pattern as the other constants in the file

**Review by arhtudormorar — APPROVED** [review]

**Pull request merged**